sound/soc/sof/loader.c

```diff
@@ -192,6 +192,43 @@ static int ext_man_get_dbg_abi_info(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static int ext_man_get_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_config_data *config =
+		container_of(hdr, struct sof_ext_man_config_data, hdr);
+	int elems_counter;
+	int elems_size;
+	int i;
+
+	/* calculate elements counter */
+	elems_size = config->hdr.size - sizeof(struct sof_ext_man_elem_header);
+	elems_counter = elems_size / sizeof(struct sof_config_elem);
+
+	dev_dbg(sdev->dev, "ext_man_get_config_data get %d config elements\n", elems_counter);
```

Comment:
during my test I found that we may have issues here with the hdr size.

The hdr.size is defined with alignment? I think we will get some empty or garbage config elem 
```
	.hdr.elem_size = ALIGN_UP(sizeof(struct ext_man_config_data),
				  EXT_MAN_ALIGN),
```

Reply:
So maybe I change message to "up to %d config elements" ? It shouldn't be a functional problem, because this empty space is guaranteed to be zero-filled (https://stackoverflow.com/questions/16015656/are-global-variables-always-initialized-to-zero-in-c)  so each additional element will be read as EMPTY one.

Reply:
@ktrzcinx That's OK for me. @lgirdwood @fredoh9 @kv2019i any suggestion about this?
I know that we need to set the array size in FW because of XCC limitation, but for kernel I am OK with
```
/* firmware configuration information */
struct sof_ext_man_config_data {
	struct sof_ext_man_elem_header hdr;

	struct sof_config_elem elems[];
} __packed;
```
And for FW side array size, I have a RFC https://github.com/thesofproject/sof/pull/3527 to add a FIRST and LAST enum to keep the change to enum to lowest, what we need is to put new enum. @fredoh9 Also has a PR https://github.com/thesofproject/sof/issues/3507 but I think we will have more effort to add new enum.

Reply:
@xiulipan Initially I wanted to change kernel header with `struct sof_config_elem elems[];`. I got an objection/comment changing kernel header only. Strong direction is that we need to sync up with fw header. This will not be accepted.